app/src/main/java/moe/apex/rule34/image/ImageBoard.kt

```diff
@@ -60,11 +64,42 @@ interface ImageBoard {
 
 
 interface GelbooruBasedImageBoard : ImageBoard {
-    fun loadPage(tags: String, page: Int, source: ImageSource, postListKey: String? = null): List<Image> {
-        val body = RequestUtil.get(imageSearchUrl.format(tags, page)).get()
+    override fun parseImage(e: JSONObject): Image? {
+        val (fileName, fileFormat) = e.getString("image").split('.', limit = 2)
+        val fileUrl = e.getString("file_url")
+        val sampleUrl = e.optString("sample_url", "")
+        val previewUrl = e.getString("preview_url")
+        val imageWidth = e.optInt("width", 1)
+        val imageHeight = e.optInt("height", 1)
+        val aspectRatio = imageWidth.toFloat() / imageHeight.toFloat()
+
+        if (fileFormat != "jpeg" && fileFormat != "jpg" && fileFormat != "png" && fileFormat != "gif")
+            return null
+
+        val metaSource = e.getString("source").takeIf { it.isNotEmpty() }
+        val metaGroupedTags = listOf(
+            TagCategory.GENERAL.group(e.getString("tags").split(" ")),
+        )
+        val metaRating = getRatingFromString(e.getString("rating"))
+        val metaPixivId = extractPixivId(metaSource)
+        val metadata = ImageMetadata(
+            source = metaSource,
+            groupedTags = metaGroupedTags,
+            rating = metaRating,
+            pixivId = metaPixivId,
+        )
+
+        return Image(fileName, fileFormat, previewUrl, fileUrl, sampleUrl, ImageSource.R34, aspectRatio, metadata)
```

Comment:
This is causing all images from Gelbooru boards (Safebooru, Gelbooru, R34) to appear like they're from the R34 source. Please fix.